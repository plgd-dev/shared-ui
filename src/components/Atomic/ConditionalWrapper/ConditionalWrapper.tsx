import { Props } from './ConditionalWrapper.types'
import { ReactElement, FC } from 'react'

const ConditionalWrapper: FC<Props> = ({ condition, wrapper, children }) => (condition ? wrapper(children as ReactElement) : children) as ReactElement

ConditionalWrapper.displayName = 'ConditionalWrapper'

export default ConditionalWrapper
