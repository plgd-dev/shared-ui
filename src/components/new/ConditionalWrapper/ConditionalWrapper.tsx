import { Props } from './ConditionalWrapper.types'
import { ReactElement } from 'react'

const ConditionalWrapper = ({ condition, wrapper, children }: Props) => (condition ? wrapper(children as ReactElement) : children)

ConditionalWrapper.displayName = 'ConditionalWrapper'

export default ConditionalWrapper
