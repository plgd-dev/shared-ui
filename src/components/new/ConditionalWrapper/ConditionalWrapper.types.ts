import { ReactElement } from 'react'

export type Props = {
    condition: boolean
    wrapper: (c: ReactElement) => ReactElement
    children: ReactElement
}
