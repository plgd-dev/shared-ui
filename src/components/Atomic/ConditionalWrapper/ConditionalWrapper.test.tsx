import { render } from '@testing-library/react'
import ConditionalWrapper from './ConditionalWrapper'

describe('<ConditionalWrapper>', () => {
    it('render correctly - snapshot', () => {
        const { asFragment } = render(
            <div>
                <ConditionalWrapper condition={false} wrapper={(e) => <div>{e}</div>} children={<div>inner content</div>} />
                <ConditionalWrapper condition={true} wrapper={(e) => <div>{e}</div>} children={<div>inner content</div>} />
            </div>
        )

        expect(asFragment()).toMatchSnapshot()
    })
})
