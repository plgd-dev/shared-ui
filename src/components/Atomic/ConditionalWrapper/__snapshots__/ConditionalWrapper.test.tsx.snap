// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConditionalWrapper> render correctly - snapshot 1`] = `
<DocumentFragment>
  <div>
    <div>
      inner content
    </div>
    <div>
      <div>
        inner content
      </div>
    </div>
  </div>
</DocumentFragment>
`;
